plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java

```diff
@@ -838,4 +839,15 @@ public void setColumnType(ConnectorSession session, JdbcTableHandle handle, Jdbc
     {
         throw new TrinoException(NOT_SUPPORTED, "This connector does not support setting column types");
     }
+
+    @Override
+    public OptionalInt getMaxColumnNameLength(ConnectorSession session)
+    {
+        try (Connection connection = connectionFactory.openConnection(session)) {
+            return OptionalInt.of(connection.getMetaData().getMaxColumnNameLength());
```

Comment:
I'd love opinions about how to improve this. I worry that creating a connection is costly but I also notice that we do seem to call `connectionFactory.openConnection` in the query builder class as well for each call of `flushAttributesAsQuery` for example.

The only "solution" I have right now is to cache the maxColumnNameLength as a field on the client. I need to recheck what the lifecycle is of a client - IIRC it's per-transaction (i.e. practically per query).

Reply:
This limit shouldn't change at least for the duration of the session, so we definitely should cache it. `applyJoin` can be called multiple times per query.

Reply:
cached on the basejdbcclient